### DISCUSSION TITLE
Do I even need react-query when using Firestore realtime updates?

### DISCUSSION THREAD

**Discussion by cmacdonnacha** [TanStack/query#1061] [General]
I have started looking into using `react-query` for the first time as I thought it could be a good fit to keep the my local state synced with my remote state in Firestore. However, since Firestore realtime updates will only ever fire when something in Firestore changes, would using something like `react-query` be overkill? Would something like Context API be simpler?

**Comment by Glinkis** [comment]
I'm wondering kind of the same. I'm considering using `react-query` in a dashboard app at work, but we have a lot of state which needs to be synced with the server in real-time. We currently handle this with redux & websockets.

**Comment by cmacdonnacha** [comment]
Yea I feel like `reac-query` would work great for a traditional REST API + DB set up, not so much realtime updates. You can of course just use `.get()` on your Firestore collection and use `react-query` to retrieve the data (which I tried and it works) but then you lose realtime updates which is one of the biggest advantages to Firestore.

**Reply by BoavistaLudwig** [reply]
> Yea I feel like `reac-query` would work great for a traditional REST API + DB set up, not so much realtime updates. You can of course just use `.get()` on your Firestore collection and use `react-query` to retrieve the data (which I tried and it works) but then you lose realtime updates which is one of the biggest advantages to Firestore.

and, what did you eventually do/conclude?
